perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/BBMRI.java

```diff
@@ -0,0 +1,161 @@
+	}
+
+	@Override
+	public Application rejectApplication(PerunSession session, Application app, String reason) throws PerunException {
+		return app;
+	}
+
+
+	/**
+	 * Checks whether all collection IDs found in user input really exists in Perun.
+	 * If not, CantBeApproved exception is thrown.
+	 *
+	 * @param session who approves the application
+	 * @param app     application
+	 * @return application
+	 * @throws CantBeApprovedException if at least one collection ID does not exist in Perun
+	 */
+	public Application beforeApprove(PerunSession session, Application app) throws PerunException {
+		// get perun and beans from session
+		Perun perun = session.getPerun();
+		Vo vo = app.getVo();
+
+		// get all collection IDs from Perun
+		Set<String> collectionIDsInPerun = new HashSet<>();
+		Group collectionsGroup = perun.getGroupsManager().getGroupByName(session, vo, "collections");
+		for (Group group : perun.getGroupsManager().getSubGroups(session, collectionsGroup)) {
+			for (Group subgroup : perun.getGroupsManager().getSubGroups(session, group)) {
+				if ("representatives".equals(subgroup.getShortName())) {
+					Attribute collectionID = perun.getAttributesManager().getAttribute(session, subgroup, "urn:perun:group:attribute-def:def:collectionID");
+					collectionIDsInPerun.add((String) collectionID.getValue());
+				}
+			}
+		}
+
+		// get the field of application with the collections
+		Set<String> collectionIDsInApplication = getCollectionIDsFromApplication(session, app);
+
+		// get non-existing collections
+		collectionIDsInApplication.removeAll(collectionIDsInPerun);
+
+		// difference must be empty
+		if (!collectionIDsInApplication.isEmpty()) {
+			throw new CantBeApprovedException("Collections " + collectionIDsInApplication + " does not exists.", "", "", "", true);
+		}
+
+		return app;
+	}
+
+	@Override
+	public void canBeApproved(PerunSession session, Application app) throws PerunException {
+
+	}
+
+	@Override
+	public void canBeSubmitted(PerunSession session, Map<String, String> params) throws PerunException {
+
+	}
+
+	/**
+	 * Gets collection IDs from a field on the application form with short name.
+	 *
+	 * @param session
+	 * @param app
+	 * @return collection IDs set
+	 */
+	private Set<String> getCollectionIDsFromApplication(PerunSession session, Application app) throws PerunException {
+		String collectionsString = null;
+		List<ApplicationFormItemData> formData = registrar.getApplicationDataById(session, app.getId());
+		for (ApplicationFormItemData field : formData) {
+			if (BIOBANK_IDS_FIELD.equals(field.getShortname())) {
+				collectionsString = field.getValue();
+				break;
+			}
+		}
+
+		if (collectionsString == null) {
+			throw new InternalErrorException("There is no field with biobank IDs on the registration form.");
+		}
+
+		// get set of collection IDs from application
+		Set<String> collectionIDsInApplication = new HashSet<>();
+		for (String collection : collectionsString.split("[,\n ]+")) {
```

Comment:
I might not be reading regex right, but is splitting by whitespace correct ? It's not stated as allowed splitter in a ``BIOBANK_IDS_FIELD``.